src/target/spirv/spirv_support.h

```diff
@@ -0,0 +1,220 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file spirv_support
+ *
+ * \brief Utility for determining which spirv capabilities a TVM
+ * target supports.
+ */
+#ifndef TVM_TARGET_SPIRV_SPIRV_SUPPORT_H_
+#define TVM_TARGET_SPIRV_SPIRV_SUPPORT_H_
+
+#include <tvm/target/target.h>
+
+namespace tvm {
+namespace codegen {
+
+/*! \brief Represents which support a Vulkan driver has that are relevant to codegen */
+struct SPIRVSupport {
```

Comment:
One thing to keep in mind is the need of cross compilation, e.g. the machine that runs compilation may not be the machine that executes the program. So we cannot do simple live query on the driver on the same machine.

It would be good to list the some attributes of interest and register them directly in the vulkan.

Reply:
Agreed, and that is my plan for the next steps.  My plan is to have many of these parameters be added to the `TVM_REGISTER_TARGET_KIND` for vulkan, both to pass them through to the codegen, and in the future to expose them to the optimization passes.